src/services/request.service.js

```diff
@@ -220,6 +220,10 @@ const sendRequest = async (engine, requestUrl, method, authentication, proxy, da
     const basic = Buffer.from(`${authentication.username}:${decryptedPassword}`).toString('base64')
     headers.Authorization = `Basic ${basic}`
   }
+  if (authentication && (authentication.type === 'Custom')) {
```

Comment:
With eslint the structure ```
if (...) {
 ...
 } else {
    if (...) {
      ...
    }
 }
``` is not allowed. Moreover if others authentications are added the structure ```
if (...) {
 ...
 }
if (...) {
 ...
 }
```  seems better for me.

Reply:
Note `else { if () { ... } }` but just `else if () { ... }`

In this specific case, the best way could be 
```
if (authentication) {
   if (authentication.type ==='Basic') { ... }
   else if (authentication.type === 'Custom') {...}
}
```
I think this way of writing these conditions is the best for now : it will be cleaner to go for a switch when we'll have more authentication type to manage.

Reply:
My view is that if we start to have several types: 

````
switch (type) {
    case 'basic': .... 
````
is more readable to me.
JF